src/coin_test/data/composer.py

```diff
@@ -93,6 +85,25 @@ def _validate_missing_data(dataset: PriceDataset) -> bool:
         missing_index = full_index.difference(dataset.df.index)
         return len(missing_index) == 0
 
+    @staticmethod
+    def _get_start_end(
+        datasets: list[PriceDataset], length: pd.Timedelta | pd.DateOffset
+    ) -> tuple[pd.Timestamp, pd.Timestamp] | None:
+        """Get start and time for datasets."""
+        start_times = [
+            dataset.df.index[0].start_time for dataset in datasets  # type: ignore
+        ]
+        end_times = [
+            dataset.df.index[-1].end_time for dataset in datasets  # type: ignore
+        ]
+
+        start_time = max(start_times)
+        end_time = min(end_times)
+        if start_time + length > end_time:
+            return None
+
+        return start_time, start_time + length
```

Comment:
Do we want to fix to the earliest window in this range? I don't see any immediate concerns but it's worth being intentional about

Reply:
Fixing to earliest was intentional, but in hindsight it might make more sense to fix to the end. Fixing to the beginning means that we won't have lookback data for some amount of time, while fixing to the end avoids that problem